core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/ProfileCredentialsUtils.java

```diff
@@ -109,16 +109,27 @@ public Optional<AwsCredentialsProvider> credentialsProvider() {
      * @param children The child profiles that source credentials from this profile.
      */
     private Optional<AwsCredentialsProvider> credentialsProvider(Set<String> children) {
+        return credentialsProviderWithMetrics(children).map(ProviderWithMetrics::provider);
+    }
+
+    /**
+     * Internal method that returns both the credentials provider and its business metrics.
+     */
+    private Optional<ProviderWithMetrics> credentialsProviderWithMetrics(Set<String> children) {
         if (properties.containsKey(ProfileProperty.ROLE_ARN) && properties.containsKey(ProfileProperty.WEB_IDENTITY_TOKEN_FILE)) {
-            return Optional.ofNullable(roleAndWebIdentityTokenProfileCredentialsProvider());
+            return Optional.of(new ProviderWithMetrics(roleAndWebIdentityTokenProfileCredentialsProvider(),
```

Comment:
It seems we are changing ofNullable to of. Do all previous methods return non null value?

Reply:
Right, I'm trying to see if this would break things. For example, if we have code path that behaves differently based on whether the value is present or not. This would not be an issue if all previous methods return a non null value